### ISSUE TITLE
Connecting to lightwalled 0.4.9 rewinds the blocks to 850,000

### ISSUE THREAD

**Issue by nighthawk24** [zcash/ZcashLightClientKit#330]
### Describe the issue
When the light client connects to the updated lightwalletd server running v0.4.9, the iOS client rewinds the state of the wallet to block 850,000 and starts re-syncing. The re-syncing callback may also not triggered in the SDK, thus causing the Syncing state to not show in the client app.

#### If so, please list the steps to reproduce below:
1. Connect to lightwalletd service running v0.4.7
2. Sync the Zcash iOS client (master branch https://github.com/zcash/zcash-ios-wallet/commit/eeac5f6312380e9b4e27f4f322adeae102895774)
3. Connect to lightwalletd service running v0.4.9
4. Observe that the syncing animation in the client app is broken.
5. Click on Available Balance.
6. See the Blocks count to start syncing again from 850,000

### Expected behaviour
Syncing should be normal when connecting to updated lightwalletd.

### Actual behaviour + errors
Starting the client and sync state is unknown, unable to send Zcash or view the current updated balance.
Sync Error 5 may or may not be shown.

- App Version: https://github.com/zcash/zcash-ios-wallet/commit/eeac5f6312380e9b4e27f4f322adeae102895774
- iOS Version: 15.1

**Comment by pacu** [comment]
Is block 850,000 your wallet birthday?

**Comment by pacu** [comment]
@nighthawk24  Does this reproduce on Android as well?

**Comment by nighthawk24** [comment]
> @nighthawk24 Is block 850,000 your wallet birthday?

No, it is 1,153,000.

> @nighthawk24 Does this reproduce on Android as well?

No.

**Comment by pacu** [comment]
Is your Lightwalletd 0.4.9 still still syncing?
What's the version of your Zcashd node?

**Comment by nighthawk24** [comment]
> Is your Lightwalletd 0.4.9 still still syncing?
Yes. mainnet.lightwalletd.com:9067
It finished syncing after catching up to 1,500,168 blocks.

> What's the version of your Zcashd node?
4.5.1-1

**Comment by nighthawk24** [comment]
After syncing, the client errors out again.
<img width="365" alt="Screen Shot 2021-12-19 at 10 19 34 AM" src="https://user-images.githubusercontent.com/1577453/146680415-3ada8d0b-33f1-4016-b9fb-70ee5f924014.png">

**Comment by pacu** [comment]
@nighthawk24 I did the following:
1. synced my local `4.5.1-1` node. 
2. launched a `v0.4.7` lightwalletd instance 
3. synced a wallet to chaintip. 
4. Then I upgraded the lightwalletd to `v0.4.9`
5. launched the wallet again and let it sync.

the wallet synced without issues. 

I also tried the lightwalletd.com instance I couldn't get the wallet to the state you mentioned.

This is my `zcash.conf` file:

````Bash
addnode=mainnet.z.cash
testnet=0
server=1
rpcuser=XXXXXX
rpcpassword=XXXXX
rpcbind=127.0.0.1
rpcport=8232
experimentalfeatures=1
txindex=1
insightexplorer=1
````

**Comment by pacu** [comment]
Please share with us your REDACTED zcash.conf file.

**Comment by nighthawk24** [comment]
> Please share with us your REDACTED zcash.conf file.

txindex=1
insightexplorer=1
experimentalfeatures=1
rpcuser=test
mainnet=1
rpcpassword=test

**Comment by mdr0id** [comment]
@nighthawk24 Are you using our docker-compose and docker images to deploy your `zcashd` and `lighwalletd` updates?

**Comment by vamsiikrishna** [comment]
we are not using Docker / Compose. 
For `zcashd`, we use the official guide for [Debian](https://zcash.readthedocs.io/en/latest/rtd_pages/install_debian_bin_packages.html).
For `lighwalletd `, we manually compile using the Makefile provided by the project .. 
Lifecycle of it maintained through `systemd`.

**Comment by pacu** [comment]
Fixed on 0.4.11
